python/ray/train/tests/lightning_test_utils.py

```diff
@@ -22,24 +24,31 @@ def training_step(self, batch):
 
     def validation_step(self, val_batch, batch_idx):
         loss = self.forward(val_batch)
+        self.loss.append(loss)
         return {"val_loss": loss}
 
-    def validation_epoch_end(self, outputs) -> None:
-        avg_loss = torch.stack([x["val_loss"] for x in outputs]).mean()
+    def on_validation_epoch_end(self) -> None:
+        avg_loss = torch.stack(self.loss).mean()
         self.log("val_loss", avg_loss)
+        self.loss.clear()
 
     def predict_step(self, batch, batch_idx):
         return self.forward(batch)
 
     def configure_optimizers(self):
-        return torch.optim.SGD(self.parameters(), lr=0.1)
+        if self.strategy == "fsdp":
+            # Feed FSDP wrapped model parameters to optimizer
+            return torch.optim.SGD(self.trainer.model.parameters(), lr=0.1)
+        else:
+            return torch.optim.SGD(self.parameters(), lr=0.1)
```

Comment:
Is this unrelated to lightning 2.0? Or is this something you need to do differently for FSDP in 2.0+?

Reply:
I need to do it differently for FSDP in 2.0+.